core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -0,0 +1,147 @@
+            config,
+            client,
+        }
+    }
+
+    pub async fn run(self, stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+        let pool = self.pool.clone();
+        loop {
+            let mut conn = pool.connection_tagged("da_dispatcher").await.unwrap();
+
+            if *stop_receiver.borrow() {
+                tracing::info!("Stop signal received, da_dispatcher is shutting down");
+                break;
+            }
+
+            if let Err(err) = self.dispatch(&mut conn).await {
+                tracing::warn!("dispatch error {err:?}");
+            }
+            if let Err(err) = self.poll_for_inclusion(&mut conn).await {
+                tracing::warn!("poll_for_inclusion error {err:?}");
+            }
+
+            drop(conn);
+            tokio::time::sleep(self.config.polling_interval()).await;
+        }
+        Ok(())
+    }
+
+    async fn dispatch(&self, conn: &mut Connection<'_, Core>) -> anyhow::Result<()> {
+        let batches = conn
+            .blocks_dal()
+            .get_ready_for_da_dispatch_l1_batches(self.config.query_rows_limit() as usize)
+            .await?;
+
+        for batch in batches {
+            let dispatch_latency = METRICS.blob_dispatch_latency.start();
+            let dispatch_response = retry(self.config.max_retries(), || {
+                self.client
+                    .dispatch_blob(batch.l1_batch_number.0, batch.pubdata.clone())
+            })
+            .await
+            .map_err(Error::msg)?;
+            dispatch_latency.observe();
+
+            conn.blocks_dal()
+                .insert_l1_batch_da(batch.l1_batch_number, dispatch_response.blob_id)
+                .await?;
+
+            METRICS
+                .last_known_l1_batch
+                .set(batch.l1_batch_number.0 as usize);
+            METRICS.blob_size.observe(batch.pubdata.len());
+            tracing::info!(
+                "Dispatched a DA for batch_number: {}",
+                batch.l1_batch_number
+            );
+        }
+
+        Ok(())
+    }
+
+    async fn poll_for_inclusion(&self, conn: &mut Connection<'_, Core>) -> anyhow::Result<()> {
+        let storage_da = conn.blocks_dal().get_da_blob_awaiting_inclusion().await?;
+
+        if let Some(storage_da) = storage_da {
+            let inclusion_data = self
+                .client
+                .get_inclusion_data(storage_da.blob_id.clone().unwrap())
+                .await
+                .map_err(Error::msg)?;
+
+            if let Some(inclusion_data) = inclusion_data {
+                conn.blocks_dal()
+                    .save_l1_batch_inclusion_data(
+                        L1BatchNumber(storage_da.l1_batch_number as u32),
+                        inclusion_data.data,
+                    )
+                    .await?;
+
+                METRICS.inclusion_latency.observe(Duration::from_secs(
+                    (Utc::now().timestamp() - storage_da.created_at.timestamp()) as u64,
```

Comment:
let's not use `created_at`/`updated_at` for anything logic related - even if just for metrics.
I understand it might look superficial, but IMO it's important - `created_at` should be only used to show **when the DB row is created**. It's like journaling. 

Imagine if you change the logic so that the rows are created before they are actually sent to DA. Thus we'd have metrics set incorrectly. 

I suggest adding a column `sent_at` and use it to report metrics.

Reply:
hmm also I think this might not work well together with the fact that we only monitor the next pending blob - what if a later one is confirmed earlier, we'll add the time of the previous one to the final reported time. This is not terribly important, but I'd change it. 

In other words, I propose to use the time when the blob was actually included (assuming this is returned from DA) as opposed to the time when we became aware that it's included

Reply:
1st point - fixed, added the `sent_at` and removed any logical dependencies on `created_at`

2nd: that would be perfect, but:
- I'm not sure that we can assume all DA layers provide that info
- If we want to poll for all dispatched blobs simultaneously to "derive that info ourselves", I'm not sure it's worth the complexity of querying for potentially 100s or 1000s of them through the APIs that don't support bulk requests, it might lead to some rate limits and increased resources consumption

but if you think that's valuable - I can spend some time looking into that